stream.go

```diff
@@ -34,99 +34,6 @@ type Stream struct {
 	connections int
 }
 
```

Comment:
Most of the changes in this file are just to make it so the options will modify a simple struct, `streamOptions`, instead of modifying the `Stream` object which contains a bunch of implementation details. PR comments indicate the real changes.